app/javascript/tagCollections/tagCollections.jsx

```diff
@@ -0,0 +1,298 @@
+import { ItemListLoadMoreButton } from '../src/components/ItemList/ItemListLoadMoreButton';
+import { ItemListCollection } from '../src/components/ItemList/ItemListCollection';
+
+const STATUS_VIEW_VALID = 'valid';
+const STATUS_VIEW_ARCHIVED = 'archived';
+const READING_LIST_ARCHIVE_PATH = '/readinglist/archive';
+const READING_LIST_PATH = '/readinglist';
+
+const FilterText = ({ selectedTags, query, value }) => {
+  return (
+    <h1>
+      {selectedTags.length === 0 && query.length === 0
+        ? value
+        : 'Nothing with this filter 🤔'}
+    </h1>
+  );
+};
+
+export class TagCollections extends Component {
+  constructor(props) {
+    super(props);
+    const { availableTags, statusView } = this.props;
+    this.state = defaultState({ availableTags, archiving: false, statusView });
+
+    this.onSearchBoxType = debounce(onSearchBoxType.bind(this), 300, {
+      leading: true,
+    });
+    this.loadNextPage = loadNextPage.bind(this);
+    this.performInitialSearch = performInitialSearch.bind(this);
+    this.search = search.bind(this);
+    this.toggleTag = toggleTag.bind(this);
+    this.clearSelectedTags = clearSelectedTags.bind(this);
+  }
+
+  componentDidMount() {
+    const { hitsPerPage, statusView } = this.state;
+
+    this.performInitialSearch({
+      containerId: 'tag-collection',
+      indexName: 'SecuredReactions',
+      searchOptions: {
+        hitsPerPage,
+        filters: `status:${statusView}`,
+      },
+    });
+  }
+
+  toggleStatusView = event => {
+    event.preventDefault();
+
+    const { query, selectedTags } = this.state;
+
+    const isStatusViewValid = this.statusViewValid();
+    const newStatusView = isStatusViewValid
+      ? STATUS_VIEW_ARCHIVED
+      : STATUS_VIEW_VALID;
+    const newPath = isStatusViewValid
+      ? READING_LIST_ARCHIVE_PATH
+      : READING_LIST_PATH;
+
+    this.setState({ statusView: newStatusView, items: [] });
+
+    this.search(query, {
+      page: 0,
+      tags: selectedTags,
+      statusView: newStatusView,
+    });
+    window.history.replaceState(null, null, newPath);
+  };
+
+  toggleArchiveStatus = (event, item) => {
+    event.preventDefault();
+
+    const { statusView, items, totalCount } = this.state;
+    window.fetch(`/reading_list_items/${item.id}`, {
+      method: 'PUT',
+      headers: {
+        'X-CSRF-Token': window.csrfToken,
+        'Content-Type': 'application/json',
+      },
+      body: JSON.stringify({ current_status: statusView }),
+      credentials: 'same-origin',
+    });
+
+    const t = this;
+    const newItems = items;
+    newItems.splice(newItems.indexOf(item), 1);
+    t.setState({
+      archiving: true,
+      items: newItems,
+      totalCount: totalCount - 1,
+    });
+    setTimeout(() => {
+      t.setState({ archiving: false });
+    }, 1000);
+  };
+
+  postCollection = e => {
+    e.preventDefault();
+    const params = {
+      name: e.target.firstChild.value,
+      tag_list: 'javascript',
```

Comment:
We will need this to be an actual form to submit a tag but glad its working,. Was it the preventdefualt that did it?